addon-manager/api/src/main/java/org/jboss/forge/addon/manager/catalog/ForgeAddonDescriptorCatalog.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Eclipse Public License version 1.0, available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.jboss.forge.addon.manager.catalog;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.jboss.forge.addon.configuration.Configuration;
+import org.jboss.forge.addon.manager.catalog.AddonDescriptor.AddonDescriptorCategory;
+import org.jboss.forge.furnace.addons.AddonRegistry;
+import org.jboss.forge.furnace.container.simple.lifecycle.SimpleContainer;
+
+/**
+ *
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+public class ForgeAddonDescriptorCatalog implements AddonDescriptorCatalog {
```

Comment:
I am not sure I agree moving this class to API. It's an impl detail tbh and I'd rather have an Abstract class with reusable parts in the API instead.